pkg/log/log_file.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main under directory cmd parses and validates user input,
+// instantiates and initializes objects imported from pkg, and runs
+// the process.
+package log
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+)
+
+const (
+	logToStdErrFlag = "logtostderr"
+	logDirFlag      = "log_dir"
+	logFileFlag     = "log_file"
+	maxSizeFlag     = "log_file_max_size"
+	maxNumFlag      = "log_file_max_num"
+
+	// Check INFO log file number every 10 mins.
+	logFileCheckInterval = time.Minute * 10
+)
+
+var (
+	logFileMaxNum = uint16(0)
+	logDir        = ""
+
+	executableName = filepath.Base(os.Args[0])
+)
+
+func AddFlags(fs *pflag.FlagSet) {
+	fs.Uint16Var(&logFileMaxNum, maxNumFlag, logFileMaxNum, "Maximum number of INFO log files to be kept. Value 0 means unlimited. Number of non-INFO log files is always unlimited")
+}
+
+// InitLogFileLimits initializes log file maximum size and maximum number limits based on the
+// command line flags.
+func InitLogFileLimits(fs *pflag.FlagSet) {
```

Comment:
I think we should let this function return an error, I don't see any reason not to, and it is itself calling functions that may return errors (even if unlikely).

Reply:
Yes, we could return errors. Just feel no big harm to let Agent continue to run even at such errors. The errors could happen only when klog implementation changes.